src/attributes.h

```diff
@@ -0,0 +1,17 @@
+// Copyright (c) 2009-2010 Satoshi Nakamoto
+// Copyright (c) 2009-2018 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#ifndef BITCOIN_ATTRIBUTES_H
+#define BITCOIN_ATTRIBUTES_H
+
+#ifdef __has_cpp_attribute
+#  if __has_cpp_attribute(nodiscard)
```

Comment:
nit: could collapse this to one level: `#if defined(__has_cpp_attribute) && __has_cpp_attribute(nodiscard)`